sql/ha_partition.cc

```diff
@@ -3190,17 +3190,21 @@ int ha_partition::read_par_file(const char *name)
     LEX_CSTRING connect_string;
     uchar buffer[4];
     char *tmp;
-    if (my_read(file, buffer, 4, MYF(MY_NABP)))
+    size_t bytes_read= my_read(file, buffer, 4, MYF(MY_NABP));
+    if (bytes_read != MY_FILE_ERROR && bytes_read > 0 )
     {
       /* No extra options; Probably not a federatedx engine */
       break;
     }
     connect_string.length= uint4korr(buffer);
     connect_string.str= tmp= (char*) alloc_root(&m_mem_root,
                                                 connect_string.length+1);
-    if (my_read(file, (uchar*) connect_string.str, connect_string.length,
-                MYF(MY_NABP)))
+
+    bytes_read= my_read(file, (uchar*) connect_string.str, connect_string.length, MYF(MY_NABP));
+    if (bytes_read != MY_FILE_ERROR && bytes_read > 0 )
+    {
```

Comment:
A bit beyond what you where changing but looks like this should be a `goto err2`?

Reply:
@grooverdan to be honest, I don't know. 
The if-block is asserting that some bytes were read from the connection arguments, so it doesn't look like it is an error to return 2.  By breaking the loop it returns 0.  Just like the other read attempt on line 3193. 
Its up to you.

Reply:
Given the other accesses on `m_connect_string` imply something is there an error jump seems prudent. Monty seems to be the only one who's touched this code recently so I'd check with him.